src/audio/dai.c

```diff
@@ -159,26 +159,40 @@ static struct comp_dev *dai_new(struct sof_ipc_comp *comp)
 
 	if (IPC_IS_SIZE_INVALID(ipc_dai->config)) {
 		IPC_SIZE_ERROR_TRACE(TRACE_CLASS_DAI, ipc_dai->config);
+		trace_dai_error("dai_new(), IPC size invalid");
 		return NULL;
 	}
 
+	tracev_dai("dai_new() after ipc size check");
+
 	dev = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM,
 		COMP_SIZE(struct sof_ipc_comp_dai));
-	if (!dev)
+	if (!dev) {
+		trace_dai_error("dai_new(), unable to allocate dev");
 		return NULL;
+	}
+
+	tracev_dai("dai_new() after dev alloc");
 
 	dai = (struct sof_ipc_comp_dai *)&dev->comp;
 	assert(!memcpy_s(dai, sizeof(*dai), ipc_dai,
 	   sizeof(struct sof_ipc_comp_dai)));
 
 	dd = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM, sizeof(*dd));
 	if (!dd) {
+		trace_dai_error("dai_new(), unable to allocate dev data");
 		rfree(dev);
 		return NULL;
 	}
 
 	comp_set_drvdata(dev, dd);
 
+	// return dev;
```

Comment:
not sure why this is commented out, is the patch based on the upstream branch?

My proposal is that the PR related to any drivers to contains the following:
* first obvious bugfixes
* new introduced functionality (it can be even a single patch) with generous commit description of the implementation.

Reply:
I will clean up the commits before setting this PR as ready for review.